### ISSUE TITLE
Make glass panes/iron bars cull each other properly when stacked vertically

### ISSUE THREAD

**Issue by muzikbike** [fxmorin/MoreCulling#153]
Vanilla issue https://bugs.mojang.com/browse/MC-50349 - this has been the case for glass panes since 1.7 and for iron bars since 1.8.

In vanilla, glass panes on top of other glass panes don't cull the other glass panes, and likewise for iron bars. This results in some rather obvious stripes in large sheets formed of the blocks when seen at the correct angles, especially for iron bars.
![2023-03-30_06 31 15](https://user-images.githubusercontent.com/52297970/228739754-75954125-a068-4114-84d0-08fdb0b8019f.png)

It's possible to manually add in cullface arguments to the models in order to have faces be culled, however a perfect solution cannot be reached through this alone. If we assign cullface only to the "post" element, we get this:
![2023-03-30_06 32 09](https://user-images.githubusercontent.com/52297970/228739916-21af076c-d282-4450-a81b-22da85cb4339.png)

This is obviously incomplete, so we'd want to have the parts extending off of the post also have their tops and bottoms culled. If we add cullface accordingly, they get culled, but another issue arises: they're culled regardless, with no regard for the geometry actually above them, resulting in faces disappearing that we don't want to disappear.
![2023-03-30_06 34 33](https://user-images.githubusercontent.com/52297970/228740261-aed7da7c-1539-4b32-b7b3-0b53155867f2.png)

In the worst case, one single post can cause four adjacent panes to not render their tops if positioned atop a cross:
![2023-03-29_19 26 01](https://user-images.githubusercontent.com/52297970/228740363-ebfdc791-cd0c-41de-905c-40f5d3d75140.png)
![2023-03-29_19 26 02](https://user-images.githubusercontent.com/52297970/228740367-795313b8-f68a-4065-b20e-d2b9e801b736.png)
![2023-03-29_19 26 25](https://user-images.githubusercontent.com/52297970/228740373-8b839bb9-ac90-40ea-81c9-fe9fd3009eaf.png)

Continuity has a built-in resource pack which implements cullface for panes, however since it's just a resource pack it's subject to all of the same issues (https://github.com/PepperCode1/Continuity/issues/74). A resource pack cannot fix this issue alone - there needs to be something that takes the actual shape of blocks above it into account, rather than simply their presence.

Here are two resource packs for testing - one which only adds cullface to posts, and another that adds it to the rest of the pane:
[pane-post-cull-only.zip](https://github.com/fxmorin/MoreCulling/files/11108158/pane-post-cull-only.zip)
[pane-all-cull.zip](https://github.com/fxmorin/MoreCulling/files/11108160/pane-all-cull.zip)

**Comment by fxmorin** [comment]
I'll attempt to add the ability to modify this cullface ;)

**Comment by muzikbike** [comment]
I haven't tested this in OptiFine yet. It may already be fixed there, or may previously have been at one point according to this issue: https://github.com/sp614x/optifine/issues/193

**Comment by muzikbike** [comment]
Indeed, OptiFine for 1.19.4 gets rid of this issue: we can clearly see the top plane of that pane still exists. This happens for Fast and Fancy connected textures. Fixing this is entirely possible.
![2023-03-30_15 30 28](https://user-images.githubusercontent.com/52297970/228888590-2a59ff42-075c-4e42-b929-1bc30fed1141.png)

Note: the weirdness of the side textures is due to multiple separate issues.

**Comment by fxmorin** [comment]
Ya I believe I know how to fix this without the optifine jank, ill try to make some time to fix it

**Comment by muzikbike** [comment]
This specific issue is now reported under https://bugs.mojang.com/browse/MC-264676.

**Comment by FxMorin** [comment]
Don't have enough time to do this, however I think if you split the top part into multiple pieces and only give a cullface to one it might work. Not sure tho

**Comment by muzikbike** [comment]
Isn't this already what vanilla does given that it uses the multipart system?
